drivers/dma/dma_stm32.c

```diff
@@ -123,7 +123,10 @@ static void dma_stm32_irq_handler(const struct device *dev, uint32_t id)
 		stream->dma_callback(dev, stream->user_data, callback_arg, DMA_STATUS_BLOCK);
 	} else if (stm32_dma_is_tc_irq_active(dma, id)) {
 #ifdef CONFIG_DMAMUX_STM32
-		stream->busy = false;
+		/* Circular buffer never stops receiving as long as peripheral is enabled */
+		if (!stream->cyclic) {
```

Comment:
is there any _event_ to disable the stream in circular buffer mode, for example is the stm32_dma_disable_stream a reason to stop the circular buffer ?

Reply:
All callers of `stm32_dma_disable_stream` already set or check the `stream->busy` flag.